### PR TITLE
[Ray] fix auto scale-in hang

### PR DESCRIPTION
**Pull request by chaokunyang**
## What do these changes do?
Fix auto scale in hang：
- [x] fix data migiration by avoiding concurrent scale in.
- [x] use `CountDownLatch` to refine `test_ownership_when_scale_in` test
- [x] use a special ray release to fix ray ownership bug
- [x] Fix `read_ray_dataset` meta error when dataset blocks are pandas DataFrame
- [x] lower ray tests timeout 

## Related issue number

Fixes #xxxx

## Check code requirements

- [ ] tests added / passed (if needed)
- [ ] Ensure all linting tests pass, see [here](https://docs.pymars.org/en/latest/development/contributing.html#check-code-styles) for how to run them

### CODE REVIEW

File: mars/services/scheduling/supervisor/autoscale.py [hunk 1]
```diff
@@ -21,6 +21,7 @@
 from collections import defaultdict
 from typing import List, Set, Dict, Optional, Any
 
+from ..errors import NoAvailableBand
```
Comment (zhongchun) [inline mars/services/scheduling/supervisor/autoscale.py:24]: This import maybe put after line 29.

**Review by qinxuye — APPROVED** [review]
LGTM

**Review by wjsi — APPROVED** [review]
LGTM

**Review by qinxuye — APPROVED** [review]
LGTM

**Pull request merged**